internal/builders/golang/build.go

```diff
@@ -192,6 +192,8 @@ func (*Builder) Build(ctx *context.Context, build config.Build, options api.Opti
 		}
 	}
 
+	addHeaderArtifactIfLibrary(ctx, build, options)
+
 	ctx.Artifacts.Add(artifact)
```

Comment:
I think that if the header thing is added.. there's no binary to add, so this can probably be removed.

I would probably make the `addHeaderArtifact...etc` return an artifact, maybe either the binary or the library, and use it here. 

also probably remove the one defined in line 138

Reply:
This is actually the intended behaviour: when building a shared library, a C header always goes with it. The header documents all the exported functions that library users may access, the `so` file provides the actual implementation.

Reply:
ah, got it... so maybe we could have both `artifacts.Header` and `artifacts.Library` or something? 

That would probably prevent some errors, like brew taps for libraries and stuff..

Reply:
Aha I see what you're saying. Yes then I guess it's reasonable to handle it separately at a higher level.

That does open up another question though! I didn't think of it before.. it would very much make sense to have this feature integrate with homebrew and nfpm, with libs and headers being deployed in the lib and include path respectively... But there again, I don't think I currently have the knowledge to make it happen..

Reply:
https://github.com/goreleaser/goreleaser/issues/3524

for now we can proceed as is. Future me will handle it :D